activesupport/lib/active_support/message_verifier.rb

```diff
@@ -53,7 +53,9 @@ def valid_message?(signed_message)
       return if signed_message.nil? || !signed_message.valid_encoding? || signed_message.blank?
 
       data, digest = signed_message.split("--".freeze)
-      data.present? && digest.present? && ActiveSupport::SecurityUtils.secure_compare(digest, generate_digest(data))
+      data.present? && digest.present? && ActiveSupport::SecurityUtils.fixed_length_secure_compare(digest, generate_digest(data))
+    rescue ArgumentError
+      false
```

Comment:
I reverted this. If this is not acceptable:
- We change to `return false unless a.bytesize == b.bytesize`. This would take us back to making it unsafe with length comparison
- Just keep this as a doc change